Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Sharing/SocketerNetworkingService.cs

```diff
@@ -0,0 +1,188 @@
+﻿using Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.Socketer;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    /// <summary>
+    /// Networking class based on Socketer's <see cref="Experimental.Socketer.NetworkConnectionManager"/>
+    /// </summary>
+    public class SocketerNetworkingService : MonoBehaviour,
+        IMatchMakingService,
+        IPlayerService,
+        INetworkingService
+    {
+        /// <summary>
+        /// TCPConnectionManager used for communicating with other devices
+        /// </summary>
+        [Tooltip("TCPConnectionManager used for communicating with other devices")]
+        [SerializeField]
+        protected TCPConnectionManager net = null;
+
+        /// <summary>
+        /// IPAddress for the user (HoloLens & HoloLens 2)
+        /// </summary>
+        [Tooltip("IPAddress for the user (HoloLens & HoloLens 2)")]
+        [SerializeField]
+        protected string userAddress = "127.0.0.1";
+
+        /// <summary>
+        /// Port for the user (HoloLens & HoloLens 2)
+        /// </summary>
+        [Tooltip("Port for the user (HoloLens & HoloLens 2)")]
+        [SerializeField]
+        protected int userPort = 7997;
+
+        /// <summary>
+        /// Timeout interval for establishing connections
+        /// </summary>
+        [Tooltip("Timeout interval for establishing connections")]
+        [SerializeField]
+        protected TimeSpan timeoutInterval = TimeSpan.Zero;
+
+        private bool runAsUser = false;
+        private Dictionary<string, string> playerIds = new Dictionary<string, string>();
+
+        /// <inheritdoc />
+        public event PlayerConnectedHandler PlayerConnected;
+
+        /// <inheritdoc />
+        public event PlayerDisconnectedHandler PlayerDisconnected;
+
+        /// <inheritdoc />
+        public event DataHandler DataReceived;
+
+        private void Start()
+        {
+#if UNITY_WSA && !UNITY_EDITOR
+            runAsUser = true;
+#elif UNITY_ANDROID || UNITY_IOS || UNITY_EDITOR
+            runAsUser = false;
```

Comment:
I guess this list could grow in the future - I guess maybe just get rid of this because we default to false? Otherwise it's a bit weird in the future where we aren't handling certain flags but really the intent is to only have runAsUser on WSA

Reply:
I will address this in a future review. We need to start enabling UNITY_WSA as a spectator in some instances in the immediate future.